controllers/controllers/workloads/cforg_controller.go

```diff
@@ -0,0 +1,200 @@
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/hierarchical-namespaces/api/v1alpha2"
+)
+
+// CFOrgReconciler reconciles a CFOrg object
+type CFOrgReconciler struct {
+	client client.Client
+	scheme *runtime.Scheme
+	log    logr.Logger
+}
+
+func NewCFOrgReconciler(client client.Client, scheme *runtime.Scheme, log logr.Logger) *CFOrgReconciler {
+	r := CFOrgReconciler{
+		client: client,
+		scheme: scheme,
+		log:    log,
+	}
+	return &r
+}
+
+const (
+	StatusConditionReady  = "Ready"
+	OrgNameLabel          = "cloudfoundry.org/org-name"
+	hierarchyMetadataName = "hierarchy"
+	APIVersion            = "workloads.cloudfoundry.org/v1alpha1"
+	Kind                  = "CFOrg"
+)
+
+//+kubebuilder:rbac:groups=workloads.cloudfoundry.org,resources=cforgs,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=workloads.cloudfoundry.org,resources=cforgs/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=workloads.cloudfoundry.org,resources=cforgs/finalizers,verbs=update
+
+//+kubebuilder:rbac:groups="",resources=namespaces,verbs=get;list;watch
+//+kubebuilder:rbac:groups=hnc.x-k8s.io,resources=subnamespaceanchors,verbs=list;create;delete;watch
+//+kubebuilder:rbac:groups=hnc.x-k8s.io,resources=hierarchyconfigurations,verbs=get;list;watch;update;patch
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the CFOrg object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.0/pkg/reconcile
+func (r *CFOrgReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	cfOrg := new(workloadsv1alpha1.CFOrg)
+	err := r.client.Get(ctx, req.NamespacedName, cfOrg)
+	if err != nil {
+		r.log.Error(err, fmt.Sprintf("Error when trying to fetch CFOrg %s/%s", req.Namespace, req.Name))
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if cfOrg.ObjectMeta.DeletionTimestamp != nil && !cfOrg.ObjectMeta.DeletionTimestamp.IsZero() {
+		return ctrl.Result{}, nil
+	}
+
+	var anchor v1alpha2.SubnamespaceAnchor
+	err = r.client.Get(ctx, req.NamespacedName, &anchor)
+	if err != nil {
+		if !k8serrors.IsNotFound(err) {
+			r.log.Error(err, fmt.Sprintf("Error getting SubnamespaceAnchor for CFOrg %s/%s", req.Namespace, req.Name))
+			return ctrl.Result{}, err
+		}
+
+		anchor, err = r.createSubnamespaceAnchor(ctx, req, cfOrg)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		// Requeue to verify subnamespaceanchor is ready
+		return ctrl.Result{RequeueAfter: 100 * time.Millisecond}, nil
+	}
+
+	namespace, isOrgReady := r.isOrgReady(ctx, anchor, cfOrg.ObjectMeta.Name)
```

Comment:
This method is weird as it performs two unrelated checks on two unrelated arguments. I would inline the anchor status check to make it more visible, and maybe extract a `getNamespace(context.Context, string) *corev1.Namespace`, which we could check for a `nil` return value. Alternatively we could make it `getNamespace(context.Context, string) (*corev1.Namespace, bool)` and use the typical Golang pattern, similarly to what we're doing now:
```golang
namespace, ok := r.getNamespace(ctx, cfOrg.ObjectMeta.Name)
if !ok {
    // ...
}
```

Reply:
Done